server/test/repository/ApplicationEventRepositoryTest.java

```diff
@@ -0,0 +1,53 @@
+package repository;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.common.collect.ImmutableList;
+import models.Account;
+import models.Applicant;
+import models.Application;
+import models.ApplicationEvent;
+import models.Program;
+import org.junit.Before;
+import org.junit.Test;
+import services.application.ApplicationEventDetails;
+import services.application.ApplicationEventDetails.StatusEvent;
+
+public class ApplicationEventRepositoryTest extends ResetPostgres {
+
+  private ApplicationEventRepository repo;
+
+  @Before
+  public void setUp() {
+    repo = instanceOf(ApplicationEventRepository.class);
+  }
+
+  @Test
+  public void insertAndGet() {
+    Program program = resourceCreator.insertActiveProgram("Program");
+    Account actor = resourceCreator.insertAccount();
+    Applicant applicant = resourceCreator.insertApplicant();
+    Application application = resourceCreator.insertActiveApplication(applicant, program);
+
+    ApplicationEventDetails details =
+        ApplicationEventDetails.builder()
+            .setEventType(ApplicationEventDetails.Type.STATUS_CHANGE)
+            .setStatusEvent(
+                StatusEvent.builder().setStatusText("Status").setEmailSent(true).build())
+            .build();
+    ApplicationEvent event =
+        new ApplicationEvent(
+            application, actor, ApplicationEventDetails.Type.STATUS_CHANGE, details);
+    repo.insert(event).toCompletableFuture().join();
```

Comment:
Mind asserting on the return value here? One interesting thing to assert in my opinion is that the event doesn't have an ID before and does have an ID after.

Reply:
done I think.

I expanded the test a bit and broke out insert and get since I realized the insert return was usable alone.